### ISSUE TITLE
Implement custom model selection for BERT training

### ISSUE THREAD

**Issue by FrancescoCasalegno** [BlueBrain/Search#290]
When pre-training/fine-tuning a model on a Masked Language Model task, by default the `huggingface/transformers` library:

- uses the loss on the evaluation dataset to select the best model,
- considers only the checkpoints available at the end of the training to do so.

We would like to be able to:
- [ ] use the mean of the Spearman scores on a test dataset of choice (e.g. BIOSSES) to select the best model
- [ ] considers all the checkpoints to do so, even if they are not all kept during training
